service/pure/src/main.rs

```diff
@@ -59,7 +61,8 @@ pub(crate) fn create_config() {
     if fs::metadata(CONFIG_FILE_PATH).is_err() {
         write_config(Config {
             timeout: DEFAULT_TIMEOUT_SEC,
-            multiaddr: None,
+            external_multiaddr: None,
+            local_multiaddr: DEFAULT_LOCAL_MULTIADDR.to_string(),
```

Comment:
do you really need `String` here? Seems like `&str` would do and you'd avoid calling `to_string`